content/blog/teach-tidyverse-2021/index.Rmd

```diff
@@ -0,0 +1,437 @@
+    You should avoid teaching functions that are deprecated and correct their usage in your students' code by suggesting the preferred alternative.
+
+-   <img src="lifecycle-superseded.svg" alt="Superseded" style="vertical-align:middle"/> Superseded indicates that there is a known better alternative for the function, but it's not going away.
+    Some examples include the following:
+
+    -   `tidyr::spread()` / `tidyr::gather()` vs. `tidyr::pivot_longer()` / `tidyr::pivot_wider()`
+
+    -   Scoped verbs (e.g., `dplyr::mutate_if()`, `dplyr::select_at()`, `dplyr::rename_all()`, etc.) vs. `dplyr::across()`
+
+    -   `dplyr::sample_n()` / `dplyr::sample_frac()` vs. `dplyr::slice_sample()` with `n` and `prop` arguments
+
+    I would recommend not teaching superseded functions to new learners, and for learners who might be aware of them already, I would recommend discouraging their use (though not correcting, i.e., no point deductions on a formative assessment), and suggesting an alternative.
+
+-   <img src="lifecycle-experimental.svg" alt="Experimental" style="vertical-align:middle"/> Experimental functions are made available so the community can try them out and provide feedback, however they come with no promises for long term stability.
+    Some examples include the following:
+
+    -   `dplyr::group_trim()` to drop unused levels of all factors that are used as grouping variables
+
+    -   in `dplyr::summarize()`: `.groups` argument to define the grouping structure of the result
+
+    -   in `dplyr::mutate()`: `.before` and `.after` arguments to control where new columns should appear
+
+    I would recommend teaching experimental functions with caution, particularly to new learners with whom you might not formally discuss the concept of a "lifecycle".
+    However there is no reason to discourage use of these functions -- if students have stumbled upon a solution that involves an experimental function or argument and has used it correctly on their own, this is likely a good indication that the experiment is working!
+
+If you'd like to learn more about the tidyverse lifecycle, I recommend the following resources:
+
+-   Blog post: [lifecycle 1.0.0](https://www.tidyverse.org/blog/2021/02/lifecycle-1-0-0/)
+-   Talk: [Maintaining the house the tidyverse built](https://www.rstudio.com/resources/rstudioglobal-2021/maintaining-the-house-the-tidyverse-built/) by Hadley Wickham at rstudio::global(2021)[^1]
+
+[^1]: I think this talk would also be a good resource for software development courses on the topic of maintaining open source software and communicating updates and changes to users.
+
+## Making reproducible examples with reprex
+
+The [**reprex**](https://reprex.tidyverse.org/) package helps users create **repr**oducible **ex**amples for posting to GitHub issues, StackOverflow, in Slack messages or snippets, or even to paste into PowerPoint or Keynote slides by placing the code to be shared in your clipboard.
```

Comment:
Again, I think this needs an early sentence like:

> I find reprex very useful when teaching because it helps my students provide me with broken code in a way that makes it as easy as possible for me to help them.